com.unity.shadergraph/Editor/Data/Nodes/Input/CustomInterpolatorNode.cs

```diff
@@ -0,0 +1,278 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEditor.Graphing;
+using UnityEditor.ShaderGraph.Internal;
+using UnityEditor.Rendering;
+
+namespace UnityEditor.ShaderGraph
+{
+    [Serializable]
+    [Title("Custom Interpolators", "Instance")]
+    class CustomInterpolatorNode : AbstractMaterialNode
+    {
+        [SerializeField]
+        internal string customBlockNodeName = "K_INVALID";
+
+        [SerializeField]
+        private BlockNode.CustomBlockType serializedType = BlockNode.CustomBlockType.Vector4;
+
+        public override bool hasPreview { get { return true; } }
+
+        internal override bool ExposeToSearcher { get => false; } // This is exposed in a special way.
+
+        internal BlockNode e_targetBlockNode // weak indirection via customBlockNodeName
```

Comment:
Nitpick, but this prefix should be changed to m_. We do not use e_ anywhere

Reply:
e_ is occasionally used for external references wherein the data is not reliably owned or defined by the object.

In the case of the node here, the BlockNode is fetched by reference name, which may or may not be valid at any given point in time (eg. if it's deleted or not deserialized yet), but may become valid in the future (eg. in post deserialization or if a new block is created with the prior name).

I don't know of any precedence for weak indirection references like this in Unity, but it needs some added dressing to make its behavior clear.